embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/capture/crash/CrashDataSourceImplTest.kt

```diff
@@ -0,0 +1,181 @@
+        localJsException = JsException("jsException", "Error", "Error", "")
+        testException = RuntimeException("Test exception")
+    }
+
+    private fun setupForHandleCrash() {
+        crashDataSource = CrashDataSourceImpl(
+            logOrchestrator,
+            sessionOrchestrator,
+            sessionProperties,
+            anrService,
+            ndkService,
+            preferencesService,
+            crashMarker,
+            logWriter,
+            configService,
+            serializer,
+            logger
+        )
+    }
+
+    @Test
+    fun `test SessionOrchestrator and LogOrchestrator are called when handleCrash is called`() {
+        //crash = CrashFactory.ofThrowable(logger, testException, null, 1)
+        setupForHandleCrash()
+
+        crashDataSource.handleCrash(testException)
+
+        assertEquals(1, anrService.forceAnrTrackingStopOnCrashCount)
+        assertEquals(1, logWriter.logEvents.size)
+        assertTrue(logOrchestrator.flushCalled)
+        assertNotNull(sessionOrchestrator.crashId)
+    }
+
+    @Test
+    fun `test LogWriter and SessionOrchestrator are called when handleCrash is called with JSException`() {
+        setupForHandleCrash()
+        crashDataSource.handleCrash(testException)
+
+        assertEquals(1, anrService.forceAnrTrackingStopOnCrashCount)
+        assertEquals(1, logWriter.logEvents.size)
+        val lastSentCrash = logWriter.logEvents.single()
+        assertEquals(logWriter.logEvents.single().schemaType.attributes()["log.record.uid"], sessionOrchestrator.crashId)
+
+        /*
+         * Verify mainCrashHandled is true after the first execution
+         * by testing that a second execution of handleCrash wont run anything
+         */
+        crashDataSource.handleCrash(testException)
+        assertEquals(1, anrService.forceAnrTrackingStopOnCrashCount)
+        assertEquals(1, logWriter.logEvents.size)
+        Assert.assertSame(lastSentCrash, logWriter.logEvents.single())
+    }
+
+    @Test
+    fun `test LogWriter and SessionOrchestrator are called when handleCrash is called with unityId`() {
+        setupForHandleCrash()
+        ndkService.lastUnityCrashId = "Unity123"
+
+        crashDataSource.handleCrash(testException)
+
+        assertEquals(1, anrService.forceAnrTrackingStopOnCrashCount)
+        assertEquals(1, logWriter.logEvents.size)
+        assertEquals(logWriter.logEvents.single().schemaType.attributes()["log.record.uid"], sessionOrchestrator.crashId)
+        assertEquals(ndkService.lastUnityCrashId, sessionOrchestrator.crashId)
+    }
+
+    @Test
+    fun `test handleCrash calls mark() method when capture_last_run config is enabled`() {
+        setupForHandleCrash()
+
+        crashDataSource.handleCrash(testException)
+
+        verify(exactly = 1) { crashMarker.mark() }
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun testSerialization() {
+        val crash = Crash(
```

Comment:
We're actually not using this class anymore in v2 - what you'd want is the `LogPayload` object that contains a crash OTel log with the appropriate attributes.